pkg/syncer/spec/suite_test.go

```diff
@@ -0,0 +1,345 @@
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+
+			// Expect downstream secret to be removed
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: secret.Name, Namespace: dataPlaneNS}, &v1.Secret{})
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+
+			// Delete gateway from control plane
+			Expect(k8sClient.Delete(ctx, createdGateway)).To(BeNil())
+			// Expect upstream gateway to be removed
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: gateway.Name, Namespace: controlPlaneNS}, &gatewayv1beta1.Gateway{})
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+
+			// Expect downstream gateway to be removed
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: gateway.Name, Namespace: dataPlaneNS}, &gatewayv1beta1.Gateway{})
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+
+		})
+		It("should ignore an annotated gateway and secret for another cluster", func() {
+			metadata.AddAnnotation(gateway, syncer.MCTC_SYNC_ANNOTATION_PREFIX+"different-cluster-id", "true")
+			Expect(k8sClient.Create(ctx, gateway)).To(BeNil())
+			createdGateway := &gatewayv1beta1.Gateway{}
+			gatewayNamespacedName := types.NamespacedName{Name: gateway.Name, Namespace: controlPlaneNS}
+
+			metadata.AddAnnotation(secret, syncer.MCTC_SYNC_ANNOTATION_PREFIX+"different-cluster-id", "true")
+			Expect(k8sClient.Create(ctx, secret)).To(BeNil())
+			createdSecret := &v1.Secret{}
+			secretNamespacedName := types.NamespacedName{Name: secret.Name, Namespace: controlPlaneNS}
+
+			// Check it does exist in the control plane
+			Expect(k8sClient.Get(ctx, gatewayNamespacedName, createdGateway)).To(BeNil())
+			Expect(k8sClient.Get(ctx, secretNamespacedName, createdSecret)).To(BeNil())
+
+			// Check gateway and secret are never synced to data plane
+			Consistently(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: gateway.Name, Namespace: dataPlaneNS}, &gatewayv1beta1.Gateway{})
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+			Consistently(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: secret.Name, Namespace: dataPlaneNS}, &v1.Secret{})
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+
+		})
+
+		It("should apply correctly annotated json mutations to a gateway and secret", func() {
+			metadata.AddAnnotation(gateway, syncer.MCTC_SYNC_ANNOTATION_PREFIX+clusterID, "true")
+			metadata.AddAnnotation(gateway, mutator.JSONPatchAnnotationPrefix+clusterID, `[
+			  {"op": "replace", "path": "/spec/gatewayClassName", "value": "istio"},
```

Comment:
Given there is currently a hardcoded transform of the `gatewayClassName` to `istio` [here](https://github.com/Kuadrant/multicluster-gateway-controller/blob/c2357b58be65b3be33210d8d81ae67823ab59399/pkg/syncer/spec/controller.go#L266), would it be better to change the value to something else?

Reply:
I actually changed it to Istio because of that hardcoded change. Once that hardcoding is removed then this could be changed. Though it should still pass then anyway.